packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -235,7 +241,7 @@ function patchFragment(n1: VFragment, n2: VFragment, parent: ParentNode, rendere
     }
 
     // Note: not reusing n1.elm, because during patching, it may be patched with another text node.
-    n2.elm = children[children.length - 1]!.elm;
+    n2.elm = children[0]!.elm;
```

Comment:
We now use the leading text anchor instead, since the other one is accessible via `nextSibling()`.

Reply:
I don't think we can reliably use `nextSibling()` to find the trailing text node. There can be any number of content nodes between `lead` and `trailing`.